### ISSUE TITLE
Support half-duplex serial

### ISSUE THREAD

**Issue by llamington** [stm32-rs/stm32f4xx-hal#592]
Currently the `serial` function requires a tuple of RX/TX pins. The [reference manual](https://www.st.com/resource/en/reference_manual/rm0390-stm32f446xx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf) states that "the TX and RX lines are internally connected", so only one pin would be required as an argument.

**Comment by burrbull** [comment]
You mean both read and write on TX line?

**Comment by llamington** [comment]
Yes that's correct

**Comment by burrbull** [comment]
Not implemented yet.

It's better to create independent `SerialHalfDuplex` structure for this with `new::(pin: TX)` and its own methods.
As I understand it will be semantically incompatible with `read`/`write` from `embedded-hal`.

**Comment by bluespider42** [comment]
@llamington Did you ever make any progress on this?

**Comment by llamington** [comment]
@bluespider42 no, unfortunately I didn't (I ended up using the STM32HAL library in C)

**Comment by bluespider42** [comment]
embassy-stm32 hal has a half-duplex uart.

**Comment by bluespider42** [comment]
https://github.com/stm32-rs/stm32f4xx-hal/pull/781
